securedrop/tests/test_journalist.py

```diff
@@ -1582,6 +1582,59 @@ def test_render_locales(config, journalist_app, test_journo, test_source):
     assert url_end + '?l=en_US' in text, text
 
 
+def test_download_selected_submissions_from_source(journalist_app,
+                                                   test_journo,
+                                                   test_source):
+    source = Source.query.get(test_source['id'])
+    submissions = utils.db_helper.submit(source, 4)
```

Comment:
I think this pattern is preferable because if the underlying object was changed during the request, SQLAlchemy won't know about it and will give us stale values. There were some tests I could only get to pass in the past by doing a refetch.